### PR TITLE
3P Media: Update label for search results

### LINKED ISSUE #3903: More appropriate label than "Trending" for search results
## Feature Description

Right now we show "Trending" above the media list when not searched at all and when performed a search. For a selected category we show its name. It might be confusing to the user to show "Trending" also for search results.

Also, when a search is made we should show something like `Results for "cat"`

### PR DESCRIPTION
**Pull request by BrookeGraham**
## Context

In `3pMedia` tab, it can be confusing what is being displayed after entering a search term. The subheader above the results displays `Trending` no matter if you have searched or not. 

## Summary

When searching for 3pMedia, we want display the subheading just above the results with the text `Search Results` since this more inline with what is being displayed. As opposed to showing a subheading of `Trending`. Category name still takes precedent when filtered by a category. 

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. In 3pMedia tab, enter a search term. You should see `Search Results` instead of the `Trending` text. 
2. If you select a category to filter by, the subheading should be the Category text. ie. "Wallpapers`

### Does this PR have a security-related impact?

no
### Does this PR change what data or activity we track or use?

no
### Does this PR have a legal-related impact?

no
## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #3903

### CODE REVIEW

File: packages/story-editor/src/components/library/panes/media/media3p/test/media3pPane.js [hunk 1]
```diff
@@ -241,6 +253,7 @@ describe('Media3pPane', () => {
   it('should render <Media3pPane /> with the category display name when selected', () => {
     useMediaResult.media3p.PROVIDER_1.state.isMediaLoaded = true;
     useMediaResult.media3p.PROVIDER_1.state.media = MEDIA;
+    useMediaResult.searchTerm = '';
```
Comment (BrookeGraham) [inline packages/story-editor/src/components/library/panes/media/media3p/test/media3pPane.js:256]: I just copied over the patterns used in the rest of this test. Because I set `useMediaResult.searchTerm = 'cats';` I had to reset it to an empty string here. I had thought that the 
```beforeEach(() => {
    useMediaResult = DEFAULT_USE_MEDIA_RESULT;
  });
```
on line 148 would have reset it to be empty. BUT that doesn't seem to be the case.

Comment (BrookeGraham) [reply packages/story-editor/src/components/library/panes/media/media3p/test/media3pPane.js:256]: Let me know if y'all think I should make this better. 🤷🏻‍♀️

File: packages/story-editor/src/components/library/panes/media/media3p/providerPanel.js [hunk 2]
```diff
@@ -114,7 +117,7 @@ function ProviderPanel({ providerType, isActive, searchTerm, ...rest }) {
         data-testid={'media-subheading'}
         shouldDisplay={shouldDisplayMediaSubheading}
       >
-        {displayName}
+        {searchTerm.length ? searchResultsLabel : displayName}
```
Comment (BrittanyIRL) [inline packages/story-editor/src/components/library/panes/media/media3p/providerPanel.js]: 🤝

File: packages/story-editor/src/components/library/panes/media/media3p/test/media3pPane.js (hunk 1 above)
Comment (BrittanyIRL) [reply packages/story-editor/src/components/library/panes/media/media3p/test/media3pPane.js:256]: think it's fine as is!

**Review by BrittanyIRL — COMMENTED** [review]
An edge case for sure, but if i have a search term and then i select a filter (like Wallpapers) then the text should update to "Wallpapers" not remain "Search Results" since it's not cactus wallpaper it's just wallpaper.

Screen Shot 2021-11-09 at 2 12 13 PM

**Comment by BrookeGraham** [comment]
> An edge case for sure, but if i have a search term and then i select a filter (like Wallpapers) then the text should update to "Wallpapers" not remain "Search Results" since it's not cactus wallpaper it's just wallpaper.

Nice catch!

**Review by samwhale — APPROVED** [review]
This is really dope! Super cool how I can make a search, add a filter, then remove the filter and continue showing `Search Results`.

**Review by BrittanyIRL — APPROVED** [review]
thanks for the update!
